app/src/org/commcare/interfaces/SortableEntityAdapter.java

```diff
@@ -0,0 +1,80 @@
+package org.commcare.interfaces;
+
+import org.commcare.CommCareApplication;
+import org.commcare.cases.entity.Entity;
+import org.commcare.cases.entity.EntitySortNotificationInterface;
+import org.commcare.cases.entity.EntitySorter;
+import org.commcare.cases.entity.NodeEntityFactory;
+import org.commcare.models.AsyncNodeEntityFactory;
+import org.commcare.suite.model.Detail;
+import org.commcare.views.notifications.NotificationMessageFactory;
+import org.javarosa.core.model.instance.TreeReference;
+
+import java.util.List;
+
+/**
+ * Created by amstone326 on 5/24/17.
+ */
+
+public abstract class SortableEntityAdapter implements EntitySortNotificationInterface {
```

Comment:
Would be great if this class could live in `commcare-core` - looks like the `notifyBadFilter` override is the only android dependency